### PR TITLE
add funcs from job_data and hook in

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
tested on rig

### CODE REVIEW

File: src/asmcnc/job/job_data.py [hunk 1]
```diff
@@ -405,3 +409,27 @@ def post_job_data_update_post_send(self):
...
+            if 'G0' in line:
+                self.g0_lines.append(i)
+                for j, next_line in enumerate(gcode[i + 1:]):
+                    if 'G1' in next_line or 'G2' in next_line or 'G3' in next_line:
+                        break
+                    else:
+                        self.g0_lines.append(i + j + 1)
+        return self.g0_lines
+
+    def get_spindle_speeds(self, gcode):
```
Comment (LettyYeti) [inline src/asmcnc/job/job_data.py:425]: when does this get called? Has potential to be slow and locking bc it's a for loop

File: src/asmcnc/job/job_data.py [hunk 2]
```diff
@@ -405,3 +409,27 @@ def post_job_data_update_post_send(self):
         self.post_production_notes = ''
         self.batch_number = ''
         self.percent_thru_job = 0
+
+    def get_excluded_line_numbers(self, gcode):
```
Comment (LettyYeti) [inline src/asmcnc/job/job_data.py:413]: Can this function either be renamed or commented with details of what it's doing. Also, when does it run, as it's a for loop and has potential to lock up kivy

**Review by LettyYeti — CHANGES_REQUESTED** [review]

File: src/asmcnc/job/job_data.py (hunk 2 above)
Comment (archiepeterjarvis) [reply src/asmcnc/job/job_data.py]: renamed

File: src/asmcnc/job/job_data.py (hunk 1 above)
Comment (archiepeterjarvis) [reply src/asmcnc/job/job_data.py:425]: inside run_job in serial_connection

**Pull request closed without merging**